keanu-project/src/test/java/io/improbable/keanu/vertices/generic/probabilistic/discrete/CategoricalVertexTest.java

```diff
@@ -1,32 +1,140 @@
 package io.improbable.keanu.vertices.generic.probabilistic.discrete;
 
 import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.generic.GenericTensor;
 import io.improbable.keanu.vertices.ConstantVertex;
 import io.improbable.keanu.vertices.dbl.DoubleVertex;
 import io.improbable.keanu.vertices.dbl.KeanuRandom;
 import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
 import io.improbable.keanu.vertices.dbl.probabilistic.DirichletVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.UniformVertex;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.closeTo;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import static org.hamcrest.Matchers.lessThan;
 
 import java.util.Arrays;
 import java.util.LinkedHashMap;
 import java.util.Map;
 import java.util.stream.Collectors;
 
 import static org.junit.Assert.assertEquals;
+import org.junit.rules.ExpectedException;
 
 public class CategoricalVertexTest {
     private static double epsilon = 0.01;
     private static int N = 100000;
 
     private KeanuRandom random;
 
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
     @Before
     public void setup() {
         random = new KeanuRandom(1);
     }
 
+    @Test
+    public void cantCreateCategoricalVertexIfShapeIsNotSpecifiedAndNonScalarShapesDoNotMatch() {
+        DoubleTensor t1 = DoubleTensor.create(new double[] {0., 0.5, 0.8, 0.2}, 2, 2);
+        DoubleTensor t2 = DoubleTensor.create(new double[] {0., 0.5, 0.8, 0.2}, 4, 1);
```

Comment:
`DoubleTensor t2 = t1.duplicate()`

Reply:
Not an exact duplicate, but I can reshape it.